### PR TITLE
#287 Dynamic Extensions cannot create Spring context when offline

### PR DESCRIPTION
**Pull request by todorinskiz**
## Description

1. Split `blueprint-integration` in two sub-projects with Spring 3 and Spring 5 specific implementations and dependencies, but sharing the same source and resources.
2. Added Spring xsd resources for offline use
3. Updated the Spring beans xsd reference to `classpath:` annotation
3.1. Defining the Spring xsd references with `spring.schemas` does not work, I think this is meant for Maven builds, where the XML is being copied to the jar, while Maven shading transformers are updating the Spring URLs by using the `spring.schemas` as source for the URL update. See e.g. https://stackoverflow.com/a/5742390

## Related Issue

https://github.com/xenit-eu/dynamic-extensions-for-alfresco/issues/287

## How Has This Been Tested?

Each Alfresco version was started with the Gradle task `:integration-tests:alfresco-community-xy:composeUp`.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] I have updated the changelog.

### CODE REVIEW

File: alfresco-dynamic-extensions-repo/build.gradle [hunk 1]
```diff
@@ -39,9 +39,14 @@ subprojects {
         ampLib "biz.aQute.bnd:biz.aQute.bnd:${project.ext.bndVersion}"
 
         blueprint(project(path: ":blueprint-override", configuration: "patchBlueprint${project.ext.geminiVersion.substring(0, 1)}"))
-        blueprint(project(":blueprint-integration"))
-        blueprint(project(":blueprint-integration:blueprint-integration-spring-3"))
-        blueprint(project(":blueprint-integration:blueprint-integration-spring-5"))
+
+        if(project.name.contains("-5")) {
+            ampLib("org.springframework:spring-core:${project.ext.springVersion}") { transitive = false }
```
Comment (vierbergenlars) [inline alfresco-dynamic-extensions-repo/build.gradle]: Why are we putting a spring package inside the AMP?

File: blueprint-integration/src/main/resources/META-INF/spring/extender/extender-context.xml [hunk 2]
```diff
@@ -1,12 +1,9 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <beans xmlns="http://www.springframework.org/schema/beans" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
 	xsi:schemaLocation="http://www.springframework.org/schema/beans
-http://www.springframework.org/schema/beans/spring-beans.xsd">
-
+						classpath:org/springframework/beans/factory/xml/spring-beans.xsd">
```
Comment (vierbergenlars) [inline blueprint-integration/src/main/resources/META-INF/spring/extender/extender-context.xml]: Doesn't the mapping above avoid having to change the XML files?

File: alfresco-dynamic-extensions-repo/build.gradle (hunk 1 above)
Comment (todorinskiz) [reply alfresco-dynamic-extensions-repo/build.gradle]: I thought it was needed, but apparently not.

File: blueprint-integration/src/main/resources/META-INF/spring/extender/extender-context.xml (hunk 2 above)
Comment (todorinskiz) [reply blueprint-integration/src/main/resources/META-INF/spring/extender/extender-context.xml]: Indeed, but then I need to add the Spring xsd resources in two places:
- `blueprint-integration`
- `alfresco-dynamic-extensions-repo`

And so I did

File: alfresco-dynamic-extensions-repo/build.gradle [hunk 3]
```diff
@@ -39,9 +39,13 @@ subprojects {
         ampLib "biz.aQute.bnd:biz.aQute.bnd:${project.ext.bndVersion}"
 
         blueprint(project(path: ":blueprint-override", configuration: "patchBlueprint${project.ext.geminiVersion.substring(0, 1)}"))
-        blueprint(project(":blueprint-integration"))
-        blueprint(project(":blueprint-integration:blueprint-integration-spring-3"))
-        blueprint(project(":blueprint-integration:blueprint-integration-spring-5"))
+
+        if(project.name.contains("-5")) {
```
Comment (kerkhofsd) [inline alfresco-dynamic-extensions-repo/build.gradle]: Maybe it's cleaner to use the Spring version MAJOR? (If this is possible)

**Comment by kerkhofsd** [comment]
I propose we manually trigger the integration tests for this PR before merge

**Comment by todorinskiz** [comment]
> I propose we manually trigger the integration tests for this PR before merge

Manually started Integration Tests on Jenkins

**Pull request merged**